### PR TITLE
Fixes for performance issues in sys.columns_internal() and sp_tablecollations_100

### PR DESCRIPTION
**Pull request by timchang514**
In addition, correct definition of sys.assemblies to not cause ambiguity during
Import-Export workflow.

One cause of the performance issues in catalogs was due
to an unnecessary JOIN with information_schema.columns. Another was due to
volatility classes for system functions, in this case OBJECT_ID. Encasing
the call in a subquery improves response time. Finally, a recent change to
sys.assembly_types introduced the is_nullable column, which caused a statement
to fail due to ambiguity when joining sys.assemblies with sys.assembly_types in
Import-Export Wizard. Correcting the definition of sys.assemblies removes this
ambiguity.

Task: BABEL-3111, BABEL-3278

Signed-off-by: Tim Chang <changti@amazon.com>

### Description

[Describe what this change achieves]
 
### Issues Resolved

[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql [hunk 1]
```diff
@@ -2543,6 +2543,241 @@ select
 from sys.objects s;
 GRANT SELECT ON sys.sysobjects TO PUBLIC;
 
+-- internal function in order to workaround BABEL-1597
+CREATE OR REPLACE FUNCTION sys.columns_internal()
```
Comment (surendravishnoi) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql:3266]: Please add upgrade tests for all updated functions/tables view. We should test when these objects are accesses from views/functions/procedures

Comment (timchang514) [reply contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql:3266]: Yes, I had just created this PR to be able to run the MVU tests to check upgrade script compatibility

**Review by macwoody — DISMISSED** [review]

**Review by priyansx — APPROVED** [review]
LGTM

File: contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql [hunk 2]
```diff
@@ -742,7 +742,7 @@ BEGIN
     from
         sys.spt_tablecollations_view s_tcv
     where
-        s_tcv.object_id = sys.object_id(@object)
+        s_tcv.object_id = (SELECT sys.object_id(@object))
```
Comment (JimFinnerty) [inline contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql:745]: I'm curious.  Why was this change necessary?

Comment (timchang514) [reply contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql:745]: Per Stefan: 

A root cause of the slow response time is [BABEL-1427](https://jira.rds.a2z.com/browse/BABEL-1427). This Jira puts further emphasis on evaluating volatility classes for system functions, in this case OBJECT_ID().

Performance can be improved by a workaround, altering the implementation of sys.sp_tablecollations_100 by putting the predicate with OBJECT_ID() in a subquery.

Replace:
        s_tcv.object_id = sys.object_id(@object)
With:
        s_tcv.object_id = ( SELECT sys.object_id(@object) )
This improves response time down to <10 seconds.

**Review by JimFinnerty — APPROVED** [review]
